pkg/kev/skaffold_test.go

```diff
@@ -262,4 +274,49 @@ var _ = Describe("Skaffold", func() {
 		})
 
 	})
+
+	Describe("SetBuildArtifacts", func() {
+
+		var skaffoldManifest *kev.SkaffoldManifest
+
+		Context("with detected service Dockerfiles", func() {
+
+			BeforeEach(func() {
+				skaffoldManifest = &kev.SkaffoldManifest{}
+			})
+
+			// Note, service name is derived from the Dockerfile location path
+			// example: src/myservice/Dockerfile will result in `myservice` service name
+
+			Context("and detected remote registry image names matching service name", func() {
+				analysis := &kev.Analysis{
+					Dockerfiles: []string{"src/myservice/Dockerfile"},
+					Images:      []string{"quay.io/myorg/myservice", "myservice"},
+				}
+
+				It("picks remote registry image path and sets correct Build configuration", func() {
+					skaffoldManifest.SetBuildArtifacts(analysis)
```

Comment:
More of a style thing, but can `skaffoldManifest.SetBuildArtifacts(analysis)` be wrapped in a `JustBeforeEach` block? 

This might mean you have to declare `analysis` next to `var skaffoldManifest ...` and use it in the different specs.

Reply:
yeh, multiple ways of going about this and often adding JustBeforeEach add extra code achieving the same thing. Also it's just two cases am testing here so no biggie. If it results in less / same number of statements I'll do it otherwise leave as is.